### ISSUE TITLE
VCH High Availability

### ISSUE THREAD

**Issue by malikkal** [vmware/vic#5430]
I was asked a question today; "what if my docker endpoint is unavailable?"

I could create multiple VCH within or across clusters and aggregate the resource required and let the dev folks worry how to distribute the load or use an LB approach. #5381 

However, is it possible to have HA for VCH using keepalived? OR think optimistically that vSphere HA would bring the failed VCH and ignore the temporary inconvenience. Thanks.

**Comment by chengwang86** [comment]
@malikkal We do support HA for the VCH endpoint. See our HA test here https://github.com/vmware/vic/blob/master/tests/manual-test-cases/Group5-Functional-Tests/5-4-High-Availability.robot

In this test, we deploy the VCH on a cluster with an nfsDatastore then power off the host on which the endpoint VM is deployed. After a few minutes HA would bring the endpoint VM back on another host. 

Is this what you want in your question?

**Comment by malikkal** [comment]
@chengwang86 I think you are referring to a vSphere cluster with shared datastore(s) and HA enabled, which brings up the cVMs during a host failure.

I was referring to an active-standby VCH configuration, where the secondary VCH would become active, when the primary is unavailable. Alternatively, what about FT protecting the VCH; will there be any corner cases because of the ISOs?

**Comment by corrieb** [comment]
@malikkal Yes. The story today is that HA will bring back cVMs or the control plan in the event of host failure. That is the extent of it.

The kind of HA you're asking for here seems to me to be the kind of thing you might expect from a layer above the VCH, similar to the load balancing story in the other issue. Let's say for example we were to enable SwarmKit to sit on top of multiple VCHs, would that solve both problems to your satisfaction?

**Comment by malikkal** [comment]
@corrieb Thanks. SwarmKit would be a great addition. However, if that takes time, any tactical solution (for the interim) would be a great help.
